pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/source/PulsarSource.java

```diff
@@ -66,7 +71,7 @@ public void open(Map<String, Object> config, SourceContext sourceContext) throws
         // Setup schemas
         log.info("Opening pulsar source with config: {}", pulsarSourceConfig);
         Map<String, ConsumerConfig<T>> configs = setupConsumerConfigs();
-
+        consumerExecutor = Executors.newFixedThreadPool(configs.size(),new DefaultThreadFactory(String.format("%s-%s",sourceContext.getSourceName(), sourceContext.getInstanceId())));
```

Comment:
why the pool size is `configs.size`?

Reply:
Because each task instance can consume multiple topics at the same time, each topic has independent threads, which can ensure that it does not affect each other and the performance is optimized

If the thread pool size is 1, the normal consumption of other topics will be blocked because one topic has no data.